src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Numerics;
+
+namespace System.Runtime.Intrinsics.Arm
+{
+    /// <summary>
+    /// This class provides access to the ARM SVE hardware instructions via intrinsics
+    /// </summary>
+    [Intrinsic]
+    [CLSCompliant(false)]
+    public abstract class Sve : AdvSimd
```

Comment:
what is an example of APIs that goes in `Sve` vs. `Sve.Arm64`? I thought we wanted something like `System.Runtime.Instrinsics.Arm.Sve` and `System.Runtime.Instrinsics.Arm.Sve2`?

Reply:
The general setup is that `Isa` exists to hold APIs that work on any arm platform

`Isa.Arm32` exists to hold APIs that only work on Arm32
`Isa.Arm64` exists to hold APIs that only work on Arm64 (we then have `Isa.X64` for xarch)

SVE is interesting because there are many APIs that "could" work on Arm32, but there are currently no plans (based on a prior discussion on one of the proposals) to expose that support there.

This means `Arm64` may be empty, but it must exist for usability reasons since `Sve` inherits from `AdvSimd` and otherwise `Sve.Arm64.IsSupported` would resolve to `AdvSimd.Arm64.IsSupported` and would lead users into a pit of failure, thinking it meant `Sve.IsSupported` would also be true

Reply:
ok, I didn't pay attention to `: AdvSimd` portion actually. I am wondering why `Sve` and `Sve.Arm64` do not inherit from `ArmBase` and `ArmBase.Arm64` respectively? I assume there will be a separate class for `Sve2` next to `Sve` class.

Reply:
SVE inherits from AdvSimd because it requires AdvSimd support to exist and so it makes it overall more convenient for the user

AdvSimd then inherits from ArmBase itself

Reply:
that makes sense.